browser_env/processors.py

```diff
@@ -512,229 +596,199 @@ def dfs(idx: int, obs_node_id: str, depth: int) -> str:
+                    try:
+                        original_alt = image.get_attribute("alt") or ""
+                        image_url = image.get_attribute("src")
+                        if not image_url.startswith(("http://", "https://", "www.")):
+                            image_url = urljoin(page.url, image_url)
+
+                        updated_alt = original_alt
+
+                        if image_url in self.url2caption:
+                            if self.url2caption[image_url] not in updated_alt:
+                                updated_alt = f"{updated_alt}, description: {self.url2caption[image_url]}"
+                        elif "data:image/svg" not in image_url:
+                            print(f"WARNING: {image_url} not in self.url2caption")
+
+                        if "url:" not in updated_alt:
+                            updated_alt = f"{updated_alt}, url: {image_url}"
+
+                        safe_updated_alt = json.dumps(updated_alt)
+                        image.evaluate(f"node => node.alt = {safe_updated_alt}")
+                    except Exception as e:
+                        print("L653 WARNING:", e)
+
+            if self.observation_type == "accessibility_tree_with_captioner":
+                frame_ax_trees = self.fetch_page_accessibility_tree(
+                    page,
+                    browser_info,
+                    current_viewport_only=self.current_viewport_only
+                )
+                content, obs_nodes_info = self.parse_accessibility_tree(frame_ax_trees)
+                content = self.clean_accesibility_tree(content)
+                self.obs_nodes_info = obs_nodes_info
+                self.meta_data["obs_nodes_info"] = obs_nodes_info
+            else:
+                content = ""  # Not used for SoM
+
+        return content
+
+    def process(self, page: Page) -> str:
         # get the tab info
         open_tabs = page.context.pages
         try:
             tab_titles = [tab.title() for tab in open_tabs]
             current_tab_idx = open_tabs.index(page)
             for idx in range(len(open_tabs)):
                 if idx == current_tab_idx:
-                    tab_titles[
-                        idx
-                    ] = f"Tab {idx} (current): {open_tabs[idx].title()}"
+                    tab_titles[idx] = f"Tab {idx} (current): {open_tabs[idx].title()}"
                 else:
                     tab_titles[idx] = f"Tab {idx}: {open_tabs[idx].title()}"
             tab_title_str = " | ".join(tab_titles)
         except Exception:
-            tab_title_str = " | ".join(
-                ["Tab {idx}" for idx in range(len(open_tabs))]
-            )
+            tab_title_str = " | ".join([f"Tab {idx}" for idx in range(len(open_tabs))])
 
         try:
-            browser_info = self.fetch_browser_info(page, client)
+            browser_info = self.fetch_browser_info(page)
         except Exception:
             page.wait_for_load_state("load", timeout=500)
```

Comment:
Should this be `sleep_after_execution * 1000` instead of 500?

Reply:
to fetch browser info specifically, seems like 500ms is OK.